<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1154</width>
    <height>639</height>
   </rect>
  </property>
  <property name="mouseTracking">
   <bool>true</bool>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>32</y>
      <width>1125</width>
      <height>525</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Tab 1</string>
     </attribute>
     <widget class="CamLabel" name="imgLabel">
      <property name="geometry">
       <rect>
        <x>4</x>
        <y>4</y>
        <width>640</width>
        <height>480</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>924</x>
        <y>396</y>
        <width>177</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Mouse Coord</string>
      </property>
      <widget class="QLabel" name="mouseXLabel">
       <property name="geometry">
        <rect>
         <x>8</x>
         <y>24</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>x = </string>
       </property>
      </widget>
      <widget class="QLabel" name="mouseYLabel">
       <property name="geometry">
        <rect>
         <x>8</x>
         <y>48</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>y = </string>
       </property>
      </widget>
     </widget>
     <widget class="YUVLabel" name="yuvLabel">
      <property name="geometry">
       <rect>
        <x>649</x>
        <y>9</y>
        <width>256</width>
        <height>256</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Sunken</enum>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>920</x>
        <y>8</y>
        <width>192</width>
        <height>261</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QListWidget" name="listWidget"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QPushButton" name="clearMarks">
           <property name="text">
            <string>Clear Marks</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="reset">
           <property name="text">
            <string>Reset</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>780</x>
        <y>272</y>
        <width>161</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Bot Data</string>
      </property>
      <widget class="QLabel" name="botPosLabel">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>24</y>
         <width>129</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>x, y =</string>
       </property>
      </widget>
      <widget class="QLabel" name="botAngleLabel">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>44</y>
         <width>129</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Angle =</string>
       </property>
      </widget>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>648</x>
        <y>272</y>
        <width>109</width>
        <height>52</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QRadioButton" name="camRadio">
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Camera</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="arenaRadio">
         <property name="autoFillBackground">
          <bool>true</bool>
         </property>
         <property name="text">
          <string>Arena</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QCheckBox" name="arena_checkBox">
      <property name="geometry">
       <rect>
        <x>656</x>
        <y>432</y>
        <width>161</width>
        <height>22</height>
       </rect>
      </property>
      <property name="text">
       <string>Arena Calibration</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>652</x>
        <y>456</y>
        <width>191</width>
        <height>29</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="startButton">
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="stopButton">
         <property name="text">
          <string>Stop</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>648</x>
        <y>348</y>
        <width>120</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>Visualization</string>
      </property>
      <widget class="QCheckBox" name="thresh_checkbox">
       <property name="geometry">
        <rect>
         <x>16</x>
         <y>20</y>
         <width>97</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>threshold</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QCheckBox" name="blobs_checkbox">
       <property name="geometry">
        <rect>
         <x>16</x>
         <y>40</y>
         <width>97</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>blobs</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QCheckBox" name="shapes_checkBox">
       <property name="geometry">
        <rect>
         <x>16</x>
         <y>60</y>
         <width>97</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>shapes</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>Tab 2</string>
     </attribute>
     <widget class="QLabel" name="descriptorVisualLabel">
      <property name="geometry">
       <rect>
        <x>648</x>
        <y>4</y>
        <width>311</width>
        <height>300</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="shapeColorLabel">
      <property name="geometry">
       <rect>
        <x>4</x>
        <y>500</y>
        <width>66</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>Color: </string>
      </property>
     </widget>
     <widget class="QLabel" name="imgLabel_tab2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>640</width>
        <height>480</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QLabel" name="debugLabel">
      <property name="geometry">
       <rect>
        <x>656</x>
        <y>464</y>
        <width>409</width>
        <height>17</height>
       </rect>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_4">
      <property name="geometry">
       <rect>
        <x>992</x>
        <y>344</y>
        <width>121</width>
        <height>113</height>
       </rect>
      </property>
      <property name="title">
       <string>Shape Data</string>
      </property>
      <widget class="QLabel" name="label">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>24</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Name:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>44</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Type:</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_3">
       <property name="geometry">
        <rect>
         <x>4</x>
         <y>64</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Color:</string>
       </property>
      </widget>
      <widget class="QLabel" name="shapeNameLabel">
       <property name="geometry">
        <rect>
         <x>52</x>
         <y>24</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="shapeTypeLabel">
       <property name="geometry">
        <rect>
         <x>52</x>
         <y>44</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="ShapeColorLabel">
       <property name="geometry">
        <rect>
         <x>52</x>
         <y>64</y>
         <width>66</width>
         <height>17</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>true</bool>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </widget>
     <widget class="QLabel" name="minDistLabel">
      <property name="geometry">
       <rect>
        <x>656</x>
        <y>408</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>964</x>
        <y>0</y>
        <width>180</width>
        <height>332</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QPlainTextEdit" name="plainTextEdit">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Minimum">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>29</height>
            </size>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="comboBox"/>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Train</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QListWidget" name="shapeListWidget"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QPushButton" name="deleteShapeButton">
           <property name="text">
            <string>Delete</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="deletAllShapeButton">
           <property name="text">
            <string>Delete All</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="secondaryMinDistLabel">
      <property name="geometry">
       <rect>
        <x>656</x>
        <y>436</y>
        <width>181</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>TextLabel</string>
      </property>
     </widget>
     <widget class="QGroupBox" name="groupBox_5">
      <property name="geometry">
       <rect>
        <x>648</x>
        <y>292</y>
        <width>120</width>
        <height>80</height>
       </rect>
      </property>
      <property name="title">
       <string>GroupBox</string>
      </property>
      <widget class="QRadioButton" name="primaryRadioButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>24</y>
         <width>116</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Primary</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
      <widget class="QRadioButton" name="secondaryRadioButton">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>48</y>
         <width>116</width>
         <height>22</height>
        </rect>
       </property>
       <property name="text">
        <string>Secondary</string>
       </property>
      </widget>
     </widget>
    </widget>
   </widget>
   <widget class="QLabel" name="fpsLabel">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>9</y>
      <width>201</width>
      <height>17</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="currentShapeLabel">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>553</y>
      <width>105</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QLabel" name="sensorDataLabel">
    <property name="geometry">
     <rect>
      <x>528</x>
      <y>556</y>
      <width>289</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>Sensor Data: </string>
    </property>
   </widget>
   <widget class="QLabel" name="secondaryShapeLabel">
    <property name="geometry">
     <rect>
      <x>128</x>
      <y>556</y>
      <width>145</width>
      <height>17</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1154</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>YUVLabel</class>
   <extends>QLabel</extends>
   <header location="global">yuvlabel.h</header>
  </customwidget>
  <customwidget>
   <class>CamLabel</class>
   <extends>QLabel</extends>
   <header location="global">camlabel.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
